### PR TITLE
implementation of image upload

### PR DESCRIPTION
**Pull request by AlejandroEscobarCampos**
## Quick Info

- **Feature**: Event Registration Process Enhancement
- **Impact**: Frontend and Backend
- **Affected Components**: Event creation form, image management system, event controllers

## What does this change?

This update significantly enhances the event registration and management process by allowing users to add images to their events and specify the event's privacy settings (public or private) during creation. Additionally, functionality has been enabled for users to modify or delete the image associated with an existing event, thereby offering a richer and more flexible user experience on the platform.

## Why are you changing that?

Implementing these features addresses the need to make events registered on the platform more visually appealing and gives users greater control over the privacy of their events. The ability to add, modify, and delete images makes each event more personal and relevant, while the privacy options ensure users can manage who has access to their event information.

## Merge/Deploy Checklist
- [x] Allow users to attach an image during the registration of a new event.
- [x] Include an option for users to specify whether the event is private or public during event creation.
- [x] Enable users to modify or delete the image associated with an existing event.

## How do you manually test this?

To manually test these changes, follow these steps:
1. Navigate to the event creation form on the platform.
2. Fill in the required event details, include an image, and select the desired privacy setting using the newly implemented fields.
3. Submit the form and ensure that the event is created with the correct image and privacy settings.
4. Access an existing event and use the edit option to change the image or adjust the privacy.
5. Confirm that the changes are saved correctly and are reflected in the event information.

### CODE REVIEW

**Review by MauricioNC — APPROVED** [review]

File: app/controllers/bright_events_controller.rb [hunk 1]
```diff
@@ -11,19 +11,26 @@ def new
...
+      redirect_to user_bright_events_path
     else
       flash[:error] = @event.errors.full_messages.to_sentence
       redirect_to new_bright_event_path
     end
   end
 
-  def edit
-    @event = BrightEvent.find(params[:id])
+  def edit() end
```
Comment (JuanequeX) [inline app/controllers/bright_events_controller.rb]: change this line to: 
```suggestion
  def edit; end
```

File: app/views/bright_events/_form.html.erb [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+    <%= f.radio_button :privacy, false %>
+  </div>
+  <% if action_name == 'edit' && @event.image.attached? %>
+    <div class="image-container">
+      <%= link_to purge_image_bright_event_path(@event), data: { turbo_method: :delete } do %>
+        <p>
+          <i class="fas fa-trash-alt"></i>
+           Eliminar imagen
+        </p>
+      <% end %>
```
Comment (JuanequeX) [inline app/views/bright_events/_form.html.erb]:
```suggestion
      <%= link_to purge_image_bright_event_path(@event), method: :delete, data: { turbo_method: :delete } do %>
  <i class="fas fa-trash-alt"></i>
  Eliminar imagen
<% end %>

```

File: app/views/bright_events/user_events.html.erb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        <th>Descripción</th>
+        <th>Fecha</th>
+        <th>Ubicación</th>
+        <th>Costo</th>
+        <th>Privacidad</th>
+        <th>Imagen</th>
+        <th>Editar</th>
+        <th>Ver</th>
+        <th>Eliminar</th>
+      </tr>
```
Comment (JuanequeX) [inline app/views/bright_events/user_events.html.erb]:
```suggestion
<tr>
  <% t(  :events, scope: %w[table_headers] ).each do |item| %>
    <th><%=  item %></th>
  <% end %>
</tr>
```

```
en:
  table_headers:
    events:
      title: "Título"
      description: "Descripción"
      date: "Fecha"
      location: "Ubicación"
      cost: "Costo"
      privacy: "Privacidad"
      image: "Imagen"
      edit: "Editar"
      view: "Ver"
      delete: "Eliminar"
```

Comment (JuanequeX) [reply app/views/bright_events/user_events.html.erb]: and this same structure can be applied to any textual content on your project, preferably set all the text in the locales

File: config/application.rb [hunk 4]
```diff
@@ -16,6 +16,18 @@ class Application < Rails::Application
     # Common ones are `templates`, `generators`, or `middleware`, for example.
     config.autoload_lib(ignore: %w(assets tasks))
 
+    # config/initializers/locale.rb
+
+    # Where the I18n library should search for translation files
+    I18n.load_path += Dir[Rails.root.join('lib', 'locale', '*.{rb,yml}')]
```
Comment (JuanequeX) [inline config/application.rb:22]: with this approach now is possible to use the structure in my last request. 👀

File: config/routes.rb [hunk 5]
```diff
@@ -7,9 +7,14 @@
   get "up" => "rails/health#show", as: :rails_health_check
 
   # Defines the root path route ("/")
-  root 'bright_events#index'
+  root 'home#index'
 
   get 'home/' => 'home#index'
```
Comment (JuanequeX) [inline config/routes.rb]: these two routes should be reduced to only `root 'home#index', what do you think?

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by juperameza — CHANGES_REQUESTED** [review]
attend @JuanequeX comments

File: app/controllers/bright_events_controller.rb (hunk 1 above)
Comment (farellano0) [reply app/controllers/bright_events_controller.rb]: Done!

File: app/views/bright_events/_form.html.erb (hunk 2 above)
Comment (farellano0) [reply app/views/bright_events/_form.html.erb]: Alll right.

File: app/views/bright_events/user_events.html.erb (hunk 3 above)
Comment (farellano0) [reply app/views/bright_events/user_events.html.erb]: Well done!

File: config/routes.rb (hunk 5 above)
Comment (Zama15) [reply config/routes.rb]: done :+1:

File: config/application.rb (hunk 4 above)
Comment (Zama15) [reply config/application.rb:22]: that was done in a preview commit :+1:

**Review by JuanequeX — APPROVED** [review]

**Review by juperameza — APPROVED** [review]

**Pull request merged**